lib/logstash/inputs/http_poller.rb

```diff
@@ -91,77 +73,101 @@ def run(queue)
-    @logger.error("Cannot read URL! (#{exception}/#{exception.message})", :name => name, :url => url)
+  def handle_decoded_event(queue, name, url, response, event, execution_time)
+    apply_metadata(event, name, url, response, execution_time)
+    queue << event
+  rescue StandardError, java.lang.Exception => e
+    @logger.error? && @logger.error("Error eventifying response!",
+                                    :exception => e,
+                                    :exception_message => e.message,
+                                    :name => name,
+                                    :url => url,
+                                    :response => response
+    )
   end
 
-  public
-  def client_config
-    c = {
-      connect_timeout: @connect_timeout,
-      socket_timeout: @socket_timeout,
-      request_timeout: @request_timeout,
-      follow_redirects: @follow_redirects,
-      automatic_retries: @automatic_retries,
-      pool_max: @pool_max,
-      pool_max_per_route: @pool_max_per_route,
-      cookies: @cookies,
-    }
-
-    if @proxy
-      # Symbolize keys if necessary
-      c[:proxy] = @proxy.is_a?(Hash) ?
-        @proxy.reduce({}) {|memo,(k,v)| memo[k.to_sym] = v; memo} :
-        @proxy
-    end
-
-    c[:ssl] = {}
-    if @ca_path
-      c[:ssl][:ca_file] = @ca_path
-    end
-
-    if (@truststore_path)
-      c.merge!(
-        truststore: @trust_store_path
+  private
+  def handle_failure(queue, name, url, exception)
+    event = LogStash::Event.new
+    apply_metadata(event, name, url)
+
+    event.tag("_http_request_failure")
+
+    # This is also in the metadata, but we send it anyone because we want this
+    # persisted by default, whereas metadata isn't. People don't like mysterious errors
+    event["_http_request_failure"] = {
+      "url" => url,
+      "name" => name,
+      "error" => exception.to_s
+   }
+
+    queue << event
+  rescue StandardError, java.lang.Exception => e
+      @logger.error? && @logger.error("Cannot read URL or send the error as an event!",
+                                      :exception => exception,
+                                      :exception_message => exception.message,
+                                      :name => name,
+                                      :url => url
       )
+  end
 
-      if (@truststore_password)
-        c.merge!(truststore_password: @trust_store_password)
-      end
-    end
-
-    c
+  private
+  def apply_metadata(event, name, url, response=nil, execution_time=nil)
+    return unless @metadata_target
+    event[@metadata_target] = event_metadata(name, url, response, execution_time)
   end
 
-  public
-  def client
-    @client ||= Manticore::Client.new(client_config)
+  private
+  def event_metadata(name, url, response=nil, execution_time=nil)
+    m = {
+        "name" => name,
+        "host" => @host,
+        "url" => url
+      }
+
+    m["runtime_millis"] = (execution_time*1000).to_i if execution_time
```

Comment:
My personal preference is to store values as the base unit (bytes, not megabytes; seconds, not milliseconds; meters, not kilometers; etc...)

Any gain in storing millisecond integer value instead of a floating point second value?